model/src/main/kotlin/Defect.kt

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (C) 2021 Bosch.IO GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.model
+
+import com.fasterxml.jackson.annotation.JsonInclude
+
+import java.net.URI
+import java.time.Instant
+
+/**
+ * A data class defining a model for software defects.
+ *
+ * Instances of this class are created by advisor implementations that retrieve information about known defects in
+ * packages.
+ */
+@JsonInclude(JsonInclude.Include.NON_DEFAULT)
+data class Defect(
+    /**
+     * The (external) ID of this defect. This is a string used by a concrete issue tracker system to reference this
+     * defect, such as a bug ID or ticket number.
+     */
+    val id: String,
+
+    /**
+     * The URL pointing to the source of this defect. This is typically a reference into the issue tracker system that
+     * contains this defect.
+     */
+    val url: URI,
+
+    /**
+     * A title for this defect. This is a short summary describing the problem at hand.
+     */
+    val title: String,
+
+    /**
+     * A state of the associated defect. The concrete meaning of this string depends on the source from where it was
+     * obtained, as different issue tracker systems use their specific terminology. Possible values could be *OPEN*,
+     * *IN PROGRESS*, *BLOCKED*, etc.
+     */
+    val state: String = UNKNOWN_STATE,
```

Comment:
My thinking here was that having non-null fields with a defined value for UNKNOWN would maybe slightly simplify client code because null checks are not needed. But if you prefer, nullable fields would be fine for me, too.

Reply:
Made the properties nullable and removed the default values.